maraboupy/MarabouCore.cpp

```diff
@@ -112,14 +114,14 @@ std::pair<std::map<int, double>, Statistics> solve(InputQuery inputQuery, std::s
     if(redirect.length()>0)
         output=redirectOutputToFile(redirect);
     try{
-        Engine engine;
-        if(!engine.processInputQuery(inputQuery)) return std::make_pair(ret, *(engine.getStatistics()));
+        Engine *engine = new Engine(verbosity);
```

Comment:
if you allocate dynamically, you have to delete() (in every possible branch of the execution).